binderhub/launcher.py

```diff
@@ -80,8 +83,16 @@ class Launcher(LoggingConfigurable):
                 else:
                     raise
 
+    async def get_user_data(self, username):
+        resp = await self.api_request(
+            'users/%s' % username,
+            method='GET',
+        )
+        body = json.loads(resp.body.decode('utf-8'))
+        return body
+
     def username_from_repo(self, repo_url):
```

Comment:
Maybe rename it to `unique_name_from_repo` now that it generates more than just a username.